src/components/artsy.tsx

```diff
@@ -0,0 +1,90 @@
+import * as React from "react"
+import * as sharify from "sharify"
+
+/**
+ * The currently signed-in user.
+ */
+export interface CurrentUser extends sharify.CurrentUser {}
+
+/**
+ * The Artsy specific props injected by the higher-order component produced by `ContextConsumer`.
+ *
+ * @see {@link ContextProvider}
+ * @see {@link ContextConsumer}
+ *
+ * @todo Once this PR for `rest` https://github.com/Microsoft/TypeScript/pull/13470 lands we’ll be able to not make this
+ *       optional and simply remove it from the props that a component wrapped with the `ContextConsumer` HOC accepts.
```

Comment:
Alas we have to wait a little while longer before we can make an injected prop non-optional to the wrapped component but not expose it at all to the user of the wrapped component, so leaving it optional for now.